nat-lab/tests/conftest.py

```diff
@@ -136,26 +138,28 @@ async def setup_check_interderp():
 
 
 SETUP_CHECKS = [
-    (setup_check_interderp, 30.0),
+    (setup_check_interderp, SETUP_CHECK_TIMEOUT_S, SETUP_CHECK_RETRIES),
 ]
 
 
 async def perform_setup_checks() -> bool:
-    for target, timeout in SETUP_CHECKS:
-        start_time = time.time()
-        while True:
-            if time.time() - start_time > timeout:
-                print(f"Target timeout reached for {target}().")
-                return False
+    for target, timeout, retries in SETUP_CHECKS:
+        while retries > 0:
             try:
-                await asyncio.wait_for(target(), timeout)
+                await asyncio.wait_for(asyncio.shield(target()), timeout)
                 break
             except asyncio.TimeoutError:
                 print(f"{target}() timeout, retrying...")
-                continue
+                retries -= 1
+                if retries > 0:
+                    continue
+                return False
             except Exception as e:  # pylint: disable=broad-exception-caught
                 print(f"An error occurred: {e}, retrying...")
-                continue
+                retries -= 1
+                if retries > 0:
```

Comment:
Why do we need the additional `if retries > 0:` checks if we are already in a `while retries > 0:` loop?

Would something like this be equivalent?
```python
        while retries > 0:
            try:
                await asyncio.wait_for(asyncio.shield(target()), timeout)
                break
            except asyncio.TimeoutError:
                print(f"{target}() timeout, retrying...")
            except Exception as e:
                print(f"An error occurred: {e}, retrying...")
            retries -= 1
        else:
            return False
```

Reply:
you're right, this is redundant, will change.